test/unit/lib/plugins/aws/package/lib/merge-custom-provider-resources.test.js

```diff
@@ -287,6 +287,72 @@ describe('mergeCustomProviderResources', () => {
       ).to.deep.equal(['a', 'b']);
     });
 
+    it('should merge for resources.extensions.*.DependsOn when destination is a string and source is a string', () => {
```

Comment:
Instead of those tests, let's setup one `runServerless` based test, which ensures to cover that case.

see https://github.com/serverless/serverless/tree/main/test#readme